sphinx/ext/intersphinx.py

```diff
@@ -271,237 +330,253 @@ def load_mappings(app: Sphinx) -> None:
     if any(updated):
         inventories.clear()
 
-        # Duplicate values in different inventories will shadow each
-        # other; which one will override which can vary between builds
-        # since they are specified using an unordered dict.  To make
-        # it more consistent, we sort the named inventories and then
-        # add the unnamed inventories last.  This means that the
-        # unnamed inventories will shadow the named ones but the named
-        # ones can still be accessed when the name is specified.
-        named_vals = []
-        unnamed_vals = []
-        for name, _expiry, invdata in intersphinx_cache.values():
-            if name:
-                named_vals.append((name, invdata))
-            else:
-                unnamed_vals.append((name, invdata))
-        for name, invdata in sorted(named_vals) + unnamed_vals:
-            if name:
-                inventories.named_inventory[name] = invdata
+        # old stuff, still used in the tests
```

Comment:
Since your PR would land in 8.x at earliest, we won't have the old intersphinx format. So we wouldn't need to support the old configurations where you had unnamed inventories.

See #12083.